users/views.py

```diff
@@ -440,21 +447,136 @@ def change_username_api_view(request: Request):
 
 
 @api_view(["PATCH"])
-def update_profile_api_view(request: Request):
-    user = request.user
+def update_profile_api_view(request: Request) -> Response:
+    user: User = request.user
     serializer = UserUpdateProfileSerializer(user, data=request.data, partial=True)
     serializer.is_valid(raise_exception=True)
 
+    assert isinstance(request.data, dict)
+    is_spam, reasoning = _check_for_spam(user, request.data)
```

Comment:
Am I right we're making an LLM request here and waiting till response? I'm wondering what impact it could have on production performance and current sync gunicorn configuration. As the alternative, we could have a cron job which checks new user profiles

Reply:
Or even better: we could create a new dramatiq actor and trigger it every time user changes profile

Reply:
I agree with this actually

Reply:
I was agreeing with hlib. I don't think it's worth it at this point. If it is untenable going forward, we can implement the dramatiq actor to do this. I can get you off the ground running on it if it comes down to it.